### PR TITLE
Main dude able to look up

### PR DESCRIPTION
**Pull request by Bedzior**
I might do a follow-up with looking up while running animation. 

EDIT: done 👍

### CODE REVIEW

**Comment by Bedzior** [comment]
* [ ] verify if running with head up should continue animation step in transition from running state
* [ ] verify running with head up to looking up transition

**Comment by Bedzior** [comment]
I would opt for merging this, as hung sprites occur in other transitions as well, so it's a general issue

File: src/game-loop/src/main-dude/states/MainDudeRunningLookingUpState.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+#include "logger/log.h"
+
+void MainDudeRunningLookingUpState::enter(MainDude &main_dude)
+{
+    main_dude._physics.set_max_x_velocity(MainDude::DEFAULT_MAX_X_VELOCITY);
+
+    if (main_dude._states.current == &main_dude._states.running)
+    {
+        log_info("Continuing running...");
```
Comment (dbeef) [inline src/game-loop/src/main-dude/states/MainDudeRunningLookingUpState.cpp:13]: No.

File: src/game-loop/src/main-dude/states/MainDudeRunningState.cpp [hunk 2]
```diff
@@ -2,12 +2,24 @@
...
+
 void MainDudeRunningState::enter(MainDude &main_dude)
 {
     main_dude._physics.set_max_x_velocity(MainDude::DEFAULT_MAX_X_VELOCITY);
-    main_dude._animation.start(static_cast<std::size_t>(MainDudeSpritesheetFrames::RUN_LEFT_0_FIRST),
-                               static_cast<std::size_t>(MainDudeSpritesheetFrames::RUN_LEFT_5_LAST),
-                               75, true);
+    if (main_dude._states.current == &main_dude._states.running_looking_up)
+    {
+        log_info("Continuing running from looking up...");
```
Comment (dbeef) [inline src/game-loop/src/main-dude/states/MainDudeRunningState.cpp:12]: No.

File: src/game-loop/src/main-dude/states/MainDudeRunningState.cpp [hunk 3]
```diff
@@ -2,12 +2,24 @@
 #include "main-dude/MainDude.hpp"
 #include "Input.hpp"
 
+#include "logger/log.h"
```
Comment (dbeef) [inline src/game-loop/src/main-dude/states/MainDudeRunningState.cpp:5]: No.

File: src/game-loop/src/main-dude/states/MainDudeRunningLookingUpState.cpp [hunk 4]
```diff
@@ -0,0 +1,117 @@
+#include "main-dude/states/MainDudeRunningLookingUpState.hpp"
+#include "main-dude/MainDude.hpp"
+#include "Input.hpp"
+
+#include "logger/log.h"
```
Comment (dbeef) [inline src/game-loop/src/main-dude/states/MainDudeRunningLookingUpState.cpp:5]: No.

File: src/game-loop/src/components/AnimationComponent.cpp [hunk 5]
```diff
@@ -47,3 +49,15 @@ void AnimationComponent::start(std::size_t start_frame, std::size_t end_frame, u
     _running = true;
     _finished = false;
 }
+
+void AnimationComponent::start_continued(std::size_t start_frame, std::size_t end_frame, uint32_t time_per_frame_ms, bool loop)
+{
+    _current_frame_index = start_frame + _end_frame_index - _current_frame_index;
```
Comment (dbeef) [inline src/game-loop/src/components/AnimationComponent.cpp:55]: Split this into calculating absolute current frame index (starting from 0) and adding it to the `start_frame`, or at least add a comment that this line is doing it.

File: src/game-loop/src/components/AnimationComponent.cpp [hunk 6]
```diff
@@ -47,3 +49,15 @@ void AnimationComponent::start(std::size_t start_frame, std::size_t end_frame, u
     _running = true;
     _finished = false;
 }
+
+void AnimationComponent::start_continued(std::size_t start_frame, std::size_t end_frame, uint32_t time_per_frame_ms, bool loop)
```
Comment (dbeef) [inline src/game-loop/src/components/AnimationComponent.cpp:53]: Name it `resume`.

File: src/game-loop/src/components/AnimationComponent.cpp [hunk 7]
```diff
@@ -47,3 +49,15 @@ void AnimationComponent::start(std::size_t start_frame, std::size_t end_frame, u
...
+{
+    _current_frame_index = start_frame + _end_frame_index - _current_frame_index;
+    _start_frame_index = start_frame;
+    _end_frame_index = end_frame;
+    _loop = loop;
+    assert(_time_per_frame_ms == time_per_frame_ms);
```
Comment (dbeef) [inline src/game-loop/src/components/AnimationComponent.cpp:59]: No need for assertion. Don't pass `loop` and `time_per_frame_ms`.

File: src/game-loop/src/components/AnimationComponent.cpp [hunk 8]
```diff
@@ -1,6 +1,8 @@
 #include "components/AnimationComponent.hpp"
 #include "main-dude/MainDude.hpp"
 
+#include <cassert>
```
Comment (dbeef) [inline src/game-loop/src/components/AnimationComponent.cpp:4]: As pointed below.

File: src/game-loop/src/main-dude/states/MainDudeRunningLookingUpState.cpp (hunk 4 above)
Comment (Bedzior) [reply src/game-loop/src/main-dude/states/MainDudeRunningLookingUpState.cpp]: My bad, I initially left them out and only kept locally, but they slipped through my sleepy fingers.

**Pull request merged**